### PR TITLE
feat:[SSCA-2576]: Adding spec for slsa generation step

### PR DESCRIPTION
**Pull request by humanshu2002**
Adding the schema for Slsa Generation step:

yaml for step when container type is selected :-

```
step:
  type: SlsaGeneration
  name: Generate SLSA
  identifier: SlsaGeneration_1
  spec:
    source:
      type: ecr | gcr | docker| acr | gar
      spec: 
        connector: string 
        host: string # Field name dependent on type
        projectId: string # Field name dependent on type
        image: string # Field name dependent on type -  To support backward compatibility kept name same as what is previously present in step
        tags: array # To support backward compatibility
        artifact_digest: string 
    attestation: # Will be same for all
      type: cosign
      spec:
        privateKey: account.KD_SSCA_Pvt_Key
        password: account.KD_SSCA_PWD
        connector: string
        key: string
    resources:
      limits:
      memory: 500Mi
      cpu: "0.5"
```
Yaml for step when 'others' type is selected:

```
step:
  type: SlsaGeneration
  name: Generate SLSA
  identifier: SlsaGeneration_1
  spec:
    source:
      type: Others
      spec: 
        artifact: # List of items in case of this type
          - name: string 
            digest: string 
  resources:
    limits:
    memory: 500Mi
    cpu: "0.5"
```

### CODE REVIEW

File: v0/pipeline/steps/common/cosign-provenance-attestation.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+title: CosignAttestation
+type: object
+$ref: provenance-attestation-spec.yaml
+required:
+  - privateKey
+  - password
+properties:
+  privateKey:
```
Comment (reetika04) [inline v0/pipeline/steps/common/cosign-provenance-attestation.yaml:8]: why are we not using snake case

**Comment by shash-harness** [comment]
LGTM, but let's test e2e in devspaces once, before merging.

**Review by reetika04 — APPROVED** [review]

File: v0/pipeline/steps/common/provenance-gcr-source-spec.yaml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  - host
+  - projectID
+  - imageName
+  - digest
+  properties:
+    connector:
+      type: string
+    host:
+      type: string
+    projectID:
```
Comment (reetika04) [inline v0/pipeline/steps/common/provenance-gcr-source-spec.yaml:16]: Why aren't we using snakecase?

File: v0/pipeline/steps/common/provenance-gcr-source-spec.yaml [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+      type: string
+    imageName:
```
Comment (reetika04) [inline v0/pipeline/steps/common/provenance-gcr-source-spec.yaml:18]: Why aren't we using snakecase?

**Review by shash-harness — APPROVED** [review]

File: v0/pipeline/steps/common/provenance-gcr-source-spec.yaml (hunk 2 above)
Comment (humanshu2002) [reply v0/pipeline/steps/common/provenance-gcr-source-spec.yaml:16]: to keep it backaward compatible with are older step which was synonymous with the build and push step

File: v0/pipeline/steps/common/provenance-gcr-source-spec.yaml (hunk 3 above)
Comment (humanshu2002) [reply v0/pipeline/steps/common/provenance-gcr-source-spec.yaml:18]: to keep it backaward compatible with are older step which was synonymous with the build and push step

**Review by archit-harness — APPROVED** [review]

**Pull request merged**